atlp-rwanda/phoenix-bn-backend
# Setup Continuous Integration & Quality Assurance pipeline

**What does this PR do?**

 - Configures Testing environment with Mocha, Chai and Sinon to ensure that all functionalities are tested before deployment.
- Travis CI is configured as a continuous integration service used to build and test the app.
- Configures Heroku Pipeline on Heroku
- Adds Travis CI and CodeClimate  badges  on this repo
- Configures CodeClimate for code analysis and Maintainability checking.

**Description of Task to be completed**
- All the necessary task for this PR are completed

 **How should this be manually tested?**

1 **Testing for continuous integration,code maintainability  and test coverage**

- switch to `ch-unitest-travisAndHeroku-configuration` branch and check if all badges are present
![Screenshot](https://user-images.githubusercontent.com/26524106/95991107-aa184800-0e2c-11eb-8f33-672e707ae48b.png)

2 **Testing if app is hosted on Heroku**
- open this link [https://phoenix-bn-staging.herokuapp.com//](https://phoenix-bn-staging.herokuapp.com//)

**What is the relevant Trello story?**
- [https://trello.com/c/C3AgjVc1/](https://trello.com/c/C3AgjVc1)
